app/helpers/pagelets_helper.rb

```diff
@@ -67,6 +67,64 @@ def th_content(col)
     sort col[:key], as: col[:label], default: col[:default_sort] || 'ASC'
   end
 
+  def defined_columns
+    columns = Pagelets::Manager.pagelets_at('hosts/_list', :hosts_table_column_header).map { |pt| { key: pt.opts[:key], label: pt.opts[:label], profiles: pt.profiles.map { |pr| pr.label } } if pt.opts[:key] }.compact
+    columns.each do |column|
+      column[:checked] = @selected_columns&.include?(column[:key].to_s)
+    end
+  end
+
+  def defined_categories
+    defined_columns.map { |col| col[:profiles] }.flatten.uniq
+  end
+
+  def checked_columns(category)
+    all_checked = false
+    category.each do |column|
+      if column[:checkProps][:checked]
+        all_checked = true
+      elsif all_checked
+        all_checked = nil
+        break
+      end
+    end
+    all_checked
+  end
+
+  def columns_view
+    columns = []
+    defined_categories.each do |category|
+      category_key = category.downcase
+      columns << {
+        name: category,
+        key: category_key,
+        defaultExpanded: category_key == 'general',
+        checkProps: {},
+        children: [],
+      }
+      category = columns.find { |c| c[:key] == category_key }
+      defined_columns.each do |column|
+        category[:children] << {
+          name: column[:label],
+          key: column[:key],
+          checkProps: { checked: column[:checked] },
+        }
+      end
+      category[:checkProps][:checked] = checked_columns(category[:children])
+    end
+    columns
+  end
+
+  def mount_column_selector
+    url = "#{api_users_path}/#{User.current[:id]}/table_preferences/#{auto_complete_controller_name}"
```

Comment:
```suggestion
    url = "#{api_users_path}/#{User.current[:id]}/table_preferences/#{controller_name}"
```

Could we use regular controller name?..